### PR TITLE
Implement traject based basic indexing

### LINKED ISSUE #558: [Spike] Using traject to convert a currently supported ArcLight EAD
While #557 will work to document what fields we will be going to.. this ticket aims to just setup the bare bones of using traject.

### PR DESCRIPTION
**Pull request by mejackreed**
Fixes #558 and will open up new work based off of documentation in #557 and current ArcLight codebase.

This PR adds the ability to index EAD documents using [traject](https://github.com/traject/traject).

This basic implementation will setup additional indexing work.

** Note ** This will require Solr 8.x for ArcLight adoptions.

### CODE REVIEW

File: solr/conf/schema.xml [hunk 1]
```diff
@@ -428,6 +428,9 @@
     See http://wiki.apache.org/solr/SpatialSearch
    -->
     <fieldtype name="geohash" class="solr.GeoHashField"/>
+    
+    <fieldType name="_nest_path_" class="solr.NestPathField" />
```
Comment (jrgriffiniii) [inline solr/conf/schema.xml]: 👍 I'm really excited to see this introduced.

Comment (mejackreed) [reply solr/conf/schema.xml]: Me too!



I think there might be some exciting implications for GeoBlacklight too!

File: lib/arclight/traject/ead2_config.rb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+require 'arclight/digital_object'
+require 'arclight/year_range'
+
+# rubocop:disable Style/MixinUsage
+extend TrajectPlus::Macros
+# rubocop:enable Style/MixinUsage
+
+settings do
+  provide 'nokogiri.namespaces',
+          'xmlns' => 'urn:isbn:1-931666-22-9'
```
Comment (jrgriffiniii) [inline lib/arclight/traject/ead2_config.rb]: This is not blocking and only out of curiosity, but I'm unfamiliar with  the patterns used in `traject`.  Should `urn:isbn:1-931666-22-9` be structured as constant in https://github.com/projectblacklight/arclight/blob/master/lib/arclight/custom_document.rb or https://github.com/projectblacklight/arclight/blob/master/lib/arclight.rb?

File: lib/arclight/traject/ead2_config.rb [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  to_field 'collection_sim', extract_xpath('/xmlns:ead/xmlns:archdesc/xmlns:did/xmlns:unittitle')
+
+  to_field 'extent_ssm', extract_xpath('./xmlns:did/xmlns:physdesc/xmlns:extent')
+  to_field 'abstract_ssm', extract_xpath('./xmlns:did/xmlns:abstract')
+  to_field 'scopecontent_ssm', extract_xpath('./xmlns:scopecontent/xmlns:p')
+  # to_field 'creator_ssm'
+  # to_field 'creator_ssim'
+  # to_field 'creators_ssim'
+  # to_field 'creator_sort'
+  # to_field 'collection_creator_ssm'
```
Comment (jrgriffiniii) [inline lib/arclight/traject/ead2_config.rb]: Were these `to_field` comments left intentionally?

File: lib/arclight/traject/ead2_config.rb [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  to_field 'prefercite_ssm', extract_xpath('xmlns:prefercite/*[local-name()!="head"]')
+  # to_field 'containers_ssim'
+  to_field 'bioghist_ssm', extract_xpath('xmlns:bioghist/*[local-name()!="head"]')
+  to_field 'acqinfo_ssm', extract_xpath('xmlns:acqinfo/*[local-name()!="head"]')
+  to_field 'relatedmaterial_ssm', extract_xpath('xmlns:relatedmaterial/*[local-name()!="head"]')
+  to_field 'separatedmaterial_ssm', extract_xpath('xmlns:separatedmaterial/*[local-name()!="head"]')
+  to_field 'otherfindaid_ssm', extract_xpath('xmlns:otherfindaid/*[local-name()!="head"]')
+  to_field 'altformavail_ssm', extract_xpath('xmlns:altformavail/*[local-name()!="head"]')
+  to_field 'originalsloc_ssm', extract_xpath('xmlns:originalsloc/*[local-name()!="head"]')
+  # to_field 'names_coll_ssim'
```
Comment (jrgriffiniii) [inline lib/arclight/traject/ead2_config.rb]: Please see https://github.com/projectblacklight/arclight/pull/565/files#r314842810

File: lib/arclight/traject/ead2_config.rb (hunk 3 above)
Comment (mejackreed) [reply lib/arclight/traject/ead2_config.rb]: Yes. These are essentially the todos cribbed from here: https://github.com/projectblacklight/arclight/blob/master/app/models/concerns/arclight/solr_document.rb#L14-L120

Additional fields will also need to be added from work completed in #557 

We didn't want to hold other work up by trying to implement all of the fields first.

File: lib/arclight/traject/ead2_config.rb (hunk 4 above)
Comment (mejackreed) [reply lib/arclight/traject/ead2_config.rb]: Similar comment response as previous.

File: lib/arclight/traject/ead2_config.rb (hunk 2 above)
Comment (mejackreed) [reply lib/arclight/traject/ead2_config.rb]: I'm inclined to leave it out for now. Our approach to namespaces might change so I'd like to leave open the flexibility of that possibility until we have more established conventions. 🤷‍♂

**Pull request merged**